Update kernel sources to v4.19.316

tescande opened the pull request.

tescande committed:
Restore patches from XCP-ng 8.3 main kernel RPM repo

Signed-off-by: Thierry Escande <thierry.escande@vates.tech>

tescande committed:
Updated kernel sources to v4-19-316

This also modifies needed patches, commenting out already merged ones
and unsupported features such as GFS2/DLM and vGPU.

Signed-off-by: Thierry Escande <thierry.escande@vates.tech>

stormi commented on SOURCES/kernel-x86_64.config:
```diff
@@ -647,7 +649,7 @@ CONFIG_HAVE_GENERIC_GUP=y
 CONFIG_EDD_OFF=y
 CONFIG_FIRMWARE_MEMMAP=y
-CONFIG_DELL_RBU=m
+# CONFIG_DELL_RBU is not set
```
IIRC we enabled DELL_RBU on purpose.

tescande replied:
My bad. Re-enabling it

stormi commented on SOURCES/kernel-x86_64.config:
```diff
@@ -1487,7 +1490,6 @@ CONFIG_NET_SCHED=y
 #
 # Queueing/Scheduling
 #
-CONFIG_NET_SCH_CBQ=m
```
Was this removed from 4.19.316?

tescande replied:
Yes

stormi commented on SOURCES/kernel-x86_64.config:
```diff
@@ -1500,7 +1502,6 @@ CONFIG_NET_SCH_TBF=m
 CONFIG_NET_SCH_CBS=m
 CONFIG_NET_SCH_ETF=m
 CONFIG_NET_SCH_GRED=m
-CONFIG_NET_SCH_DSMARK=m
```
Was this removed from 4.19.316?

tescande replied:
Yes

stormi commented on SOURCES/kernel-x86_64.config:
```diff
@@ -1522,14 +1523,11 @@ CONFIG_NET_SCH_PLUG=m
 #
 CONFIG_NET_CLS=y
 CONFIG_NET_CLS_BASIC=m
-CONFIG_NET_CLS_TCINDEX=m
```
Was this removed from 4.19.316?

tescande replied:
Yes

stormi commented on SOURCES/kernel-x86_64.config:
```diff
@@ -1522,14 +1523,11 @@ CONFIG_NET_SCH_PLUG=m
 CONFIG_NET_CLS_U32=m
 # CONFIG_CLS_U32_PERF is not set
 CONFIG_CLS_U32_MARK=y
-CONFIG_NET_CLS_RSVP=m
-CONFIG_NET_CLS_RSVP6=m
```
Was this removed from 4.19.316?

tescande replied:
Yes

stormi commented on SOURCES/kernel-x86_64.config:
```diff
@@ -1663,8 +1662,7 @@ CONFIG_PREVENT_FIRMWARE_BUILD=y
 #
 CONFIG_FW_LOADER=y
 CONFIG_EXTRA_FIRMWARE=""
-CONFIG_FW_LOADER_USER_HELPER=y
```
Why not set now?

tescande replied:
My bad. Re-enabling it

stormi commented on SOURCES/kernel-x86_64.config:
```diff
@@ -1722,7 +1720,6 @@ CONFIG_BLK_DEV_DRBD=m
 # CONFIG_DRBD_FAULT_INJECTION is not set
 CONFIG_BLK_DEV_NBD=m
 CONFIG_BLK_DEV_SKD=m
-CONFIG_BLK_DEV_SX8=m
```
Was this removed from 4.19.316?

tescande replied:
Yes

stormi commented on SOURCES/kernel-x86_64.config:
```diff
@@ -3971,7 +3967,7 @@ CONFIG_INFINIBAND_QIB=m
 CONFIG_INFINIBAND_QIB_DCA=y
 CONFIG_INFINIBAND_CXGB3=m
 CONFIG_INFINIBAND_CXGB4=m
-CONFIG_INFINIBAND_I40IW=m
```
Disabled on purpose?

tescande replied:
Nope, missed that one. Thanks

stormi commented on SOURCES/kernel-x86_64.config:
```diff
@@ -4334,7 +4330,7 @@ CONFIG_GENERIC_PHY=y
 #
 CONFIG_RAS=y
 # CONFIG_RAS_CEC is not set
-CONFIG_THUNDERBOLT=m
```
I think this was enabled on purpose on kernel-alt.

tescande replied:
Re-enabling it

stormi commented on SOURCES/kernel-x86_64.config:
```diff
@@ -4491,7 +4487,7 @@ CONFIG_FAT_DEFAULT_IOCHARSET="iso8859-1"
 # CONFIG_FAT_DEFAULT_UTF8 is not set
 CONFIG_NTFS_FS=m
 # CONFIG_NTFS_DEBUG is not set
-CONFIG_NTFS_RW=y
```
This was enabled on purpose.

tescande replied:
Re-enabling it

stormi commented on SPECS/kernel-alt.spec:
```suggestion
# A patch without comment means the patch applies without modification, except
```

tescande replied:
Fixed. Thanks

stormi commented on SPECS/kernel-alt.spec:
I think there may be a mix of GFS2/DLM patches and merged upstream patches? The xen-netback patches are suspicious, here.

tescande replied:
Fixed

stormi commented on SPECS/kernel-alt.spec:
Still related to GFS2?

tescande replied:
Fixed

stormi commented on SPECS/kernel-alt.spec:
Still related to GFS2?

tescande replied:
Fixed

stormi commented on SPECS/kernel-alt.spec:
Still related to vGPU?

tescande replied:
Fixed

stormi commented on SPECS/kernel-alt.spec:
Merged upstream, DLM not supported, or both?

tescande replied:
Fixed

stormi commented on SPECS/kernel-alt.spec:
All merged upstream, according to the convention. Right?

General comment: I think a `# Merged upstream` comment works better than no comment at all, because it allows to be sure of the packager's intent when commenting out the patches.

tescande replied:
Yep, agreed

stormi commented on SPECS/kernel-alt.spec:
Included in the GFS2 group?

tescande replied:
Fixed

stormi commented on SPECS/kernel-alt.spec:
Included in the GFS2 group?

tescande replied:
Fixed

stormi commented on SPECS/kernel-alt.spec:
Ok, you know what I'm going to ask here.

tescande replied:
Fixed

tescande committed:
Sync with main kernel repo v4.19.19-8.0.34.1

This synchronizes patches and spec file with the main kernel revision
v4.19.19-8.0.34.1

Signed-off-by: Thierry Escande <thierry.escande@vates.tech>

stormi commented on SPECS/kernel-alt.spec:
```diff
@@ -1062,6 +1095,7 @@ fi
 * Fri Jun 21 2024 Thierry Escande <thierry.escande@vates.tech> - 4.19.316+1-1
 - Import kernel source v4.19.316
 - Rebase and rework needed patches from main kernel repo
```
Please indicate the exact release of the main kernel we synced with. This will help know when we last synchronised. Especially now that the changelog entry date is *before* the date when the main kernel was last updated.

stormi commented:
Last call for reviews. I'd like we merged and built this tomorrow. Comments will be possible after the merge, but any fixes will then have to be done in a separate PR.

bleader reviewed (approved):
Only had a look at the config file changes and an overview of the spec file, but lgtm.

tescande committed:
Fix Requires on python3-xcp-libs

This replaces the Requires on xcp-python-libs with python3-xcp-libs.
This was not failing because python3-xcp-libs is a base package of
XCP-ng 8.3 and these are all post install requires but the wrong
package was actually installed.

Reported-by: Samuel Verschelde <stormi-xcp@ylix.fr>
Signed-off-by: Thierry Escande <thierry.escande@vates.tech>

tescande commented:
Koji build v8.3-incoming: https://koji.xcp-ng.org/buildinfo?buildID=3376

rushikeshjadhav commented:
@tescande was `git-lfs` used while commit? Im facing trouble downloading the `SOURCES/kernel-4.19.316.tar.gz` via `git pull`. I already have `git-lfs` installed.

tescande commented:
> @tescande was `git-lfs` used while commit? Im facing trouble downloading the `SOURCES/kernel-4.19.316.tar.gz` via `git pull`. I already have `git-lfs` installed.

Yes, lfs was used. What `git lfs ls-files` says ?

rushikeshjadhav commented:
> > @tescande was `git-lfs` used while commit? Im facing trouble downloading the `SOURCES/kernel-4.19.316.tar.gz` via `git pull`. I already have `git-lfs` installed.
> 
> Yes, lfs was used. What `git lfs ls-files` says ?

```
[root@92d5f9777852 rpmbuild]# git lfs ls-files
465c30cf6b * SOURCES/kernel-4.19.316.tar.gz
```

As of now I downloaded the raw package and testing it on one host.
